### PR TITLE
Add undeletetree command

### PR DESCRIPTION
**Pull request by pgporada**
While going through the steps to freeze testflume 2019 and 2020, I misstepped requiring me to undelete those shards. Thankfully you've all provided the `UndeleteTree` api call which made this far less stressful. The misstep was figuring out how to get a frozen_sth for certificate-transparency-go.

Here's how I detected that I needed this
```
$ curl -s https://testflume.ct.letsencrypt.org/2019/ct/v1/get-sth
Not Found
rpc error: code = NotFound desc = tree 7098813260217451818 not found

$ ./undeletetree --admin_server log-server:8090 --log_id=${LOG_ID}

MariaDB [database]> select TreeId,TreeState,DisplayName,Deleted from Trees;
+---------------------+-----------+-------------+---------+
| TreeId              | TreeState | DisplayName | Deleted |
+---------------------+-----------+-------------+---------+
| 2115833961257890595 | FROZEN    | 2020        |       1 |  
| 7098813260217451818 | FROZEN    | 2019        |       0 |   <==== undeleted
+---------------------+-----------+-------------+---------+
2 rows in set (0.001 sec)

$ curl -s https://testflume.ct.letsencrypt.org/2019/ct/v1/get-sth
{"tree_size":666628491,"timestamp":1614610745490,"sha256_root_hash":"ul1sABvz3nsF6GhyCPDOC2keQPQ8JWDKWIZXqqEWBcM=","tree_head_signature":"BAMARzBFAiAzRHa5zMh0g80Do0U51mGlzoK8uvbCLVgcYHfVdTKlsAIhAMnqPQ53hW9P5bnhksrxUI1yU8fwvt/lkr92bF9Oq84k"}
```

Here's what I should have done instead:

1. Run `updatetree --admin_server log-server:8090 --tree_id=${LOG_ID} -print -tree_state DRAINING`
2. Run `updatetree --admin_server log-server:8090 --tree_id=${LOG_ID} -print -tree_state FROZEN`
3. Get the STH
4. Run `deletetree --admin_server log-server:8090 --tree_id=${LOG_ID}`

### CODE REVIEW

**Comment by Martin2112** [comment]
If you're just freezing the tree then you don't need to call delete on it, it can stay in frozen state. There is a doc for this process. The draining state is to wait for integration of previous submissions to complete, new ones are blocked.

Maybe the code in this PR could be added to the `deletetree` command as a flag to avoid duplicating all the gRPC boilerplate?

**Comment by pgporada** [comment]
@Martin2112 Good idea. I've pushed a change that adds a new flag to deletetree.

**Comment by pgporada** [comment]
> If you're just freezing the tree then you don't need to call delete on it, it can stay in frozen state. There is a doc for this process. The draining state is to wait for integration of previous submissions to complete, new ones are blocked.

I followed the doc and built one tailored to our specific environment. We want to have the treeGC delete the frozen data hence the `deletetree` call. However, upon running `deletetree` I was unable to get an STH prompting me to use `undeletetree`. 

Currently our shards are frozen, not deleted. When I find a solution to https://github.com/google/certificate-transparency-go/issues/773 I will be deleting the shards.

**Review by Martin2112 — APPROVED** [review]

**Comment by Martin2112** [comment]
Not sure why the cloud build is failing. I don't seem to be able to access the logs. @paulmattei ?

File: cmd/deletetree/main.go [hunk 1]
```diff
@@ -50,8 +51,15 @@ func main() {
 	defer conn.Close()
 
 	a := trillian.NewTrillianAdminClient(conn)
-	_, err = a.DeleteTree(context.Background(), &trillian.DeleteTreeRequest{TreeId: *logID})
-	if err != nil {
-		glog.Exitf("Delete failed: %v", err)
+	if *undeleteTree == false {
```
Comment (pphaneuf) [inline cmd/deletetree/main.go:54]: There's a presubmit lint warning here, could you fix it please?

cmd/deletetree/main.go:54:5: S1002: should omit comparison to bool constant, can be simplified to `!*undeleteTree` (gosimple)

Comment (pgporada) [reply cmd/deletetree/main.go]: I've fixed this in my latest commit _many months later_.

**Comment by pgporada** [comment]
@mhutchinson Could you do a `/gcbrun` for this please?

**Comment by mhutchinson** [comment]
This is failing cloudbuild with errors unrelated to this change:
```
Running: kubectl apply --server-dry-run -f=examples/deployment/kubernetes/etcd-deployment.yaml
Error: unknown flag: --server-dry-run
```

This appears to be https://github.com/Praqma/helmsman/issues/607, which is old, but does align with the time this PR was first created.

@pgporada can you rebase this and we'll try again?

**Comment by mhutchinson** [comment]
Actually it's even simpler than that. It's just that the flag format was actually changed in this repo to work around the change upstream: https://github.com/google/trillian/commit/54802f7cb76eced76d02a835bb3714cfaf506a80#diff-315aa47a29f0963114c0520aa5fa004a015f4252173eaaca74f14816331a2915. Once we have that change in this PR we should be golden.

**Comment by mhutchinson** [comment]
#2638 is merged. Going to close this one now. Thanks for adding this Phil :-)

**Pull request closed without merging**